Framework/Algorithms/test/Q1DWeightedTest.h

```diff
@@ -265,6 +326,137 @@ class Q1DWeightedTest : public CxxTest::TestSuite {
+       << centerX << "\t" << centerY;
+    // the other parameters are omitted, since they are not useful for us
+    return ss.str();
+  }
+
+  std::string createDummyViewport(double transX, double transY, double zoom, double rotation0, double rotation1,
+                                  double rotation2, double rotation3) {
+    std::stringstream ss;
+    ss << "Translation\t" << transX << "\t" << transY << std::endl;
+    ss << "Zoom\t" << zoom << std::endl;
+    ss << "Rotation\t" << rotation0 << "\t" << rotation1 << "\t" << rotation2 << "\t" << rotation3;
+    return ss.str();
+  }
+
+  void populateAlgorithm(std::string outputWS, std::string wedgesWS, bool useTable, bool asymmetric = false,
+                         std::string binning = "0.001,0.001,0.08") {
+    std::string asymm_flag = asymmetric ? "1" : "0";
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("InputWorkspace", m_inputWS))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputWorkspace", outputWS))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("OutputBinning", binning))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeWorkspace", wedgesWS))
+    TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("AsymmetricWedges", asymm_flag))
+
+    if (useTable) {
+      TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("ShapeTable", "MaskShapes"))
+    } else {
+      TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("NumberOfWedges", "2"))
+      TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeAngle", "90"))
+      TS_ASSERT_THROWS_NOTHING(radial_average.setPropertyValue("WedgeOffset", "0"))
+    }
+  }
+
+  void compareWorkspaces(std::string refWS, std::string toCompare) {
+    WorkspaceGroup_sptr result;
+    TS_ASSERT_THROWS_NOTHING(
+        result = std::dynamic_pointer_cast<WorkspaceGroup>(AnalysisDataService::Instance().retrieve(toCompare)))
+
+    WorkspaceGroup_sptr ref =
+        std::dynamic_pointer_cast<WorkspaceGroup>(AnalysisDataService::Instance().retrieve(refWS));
+
+    TS_ASSERT_EQUALS(result->getNumberOfEntries(), ref->getNumberOfEntries())
+    std::string tolerance = "1e-12";
+
+    CompareWorkspaces comparison;
+    comparison.initialize();
+
+    for (int i = 0; i < result->getNumberOfEntries(); i++) {
+      auto resultWS = std::dynamic_pointer_cast<MatrixWorkspace>(result->getItem(i));
+      TS_ASSERT(resultWS)
+      auto refWS = std::dynamic_pointer_cast<MatrixWorkspace>(ref->getItem(i));
+
+      comparison.setProperty("Workspace1", refWS);
+      comparison.setProperty("Workspace2", resultWS);
+      comparison.setPropertyValue("Tolerance", tolerance);
+      comparison.setPropertyValue("CheckAllData", "1");
+      comparison.setPropertyValue("CheckType", "1");
+      comparison.setPropertyValue("ToleranceRelErr", "1");
+      comparison.execute();
```

Comment:
CompareWorkspaces returns a boolean, you have to assert on that, just running it is not enough.

Reply:
Moreover, if you run compare workspaces on the workspace group, it will compare both the order, and the workspaces pairwise.